pkg/restore/service_action.go

```diff
@@ -101,3 +83,29 @@ func getPreservedPorts(obj runtime.Unstructured) (map[string]bool, error) {
 	}
 	return preservedPorts, nil
 }
+
+func deleteNodePorts(obj runtime.Unstructured, spec *map[string]interface{}) error {
+	preservedPorts, err := getPreservedPorts(obj)
+	if err != nil {
+		return err
+	}
+
+	ports, err := collections.GetSlice(obj.UnstructuredContent(), "spec.ports")
+	serviceType, _ := collections.GetString(*spec, "type")
+	if err != nil && serviceType != "ExternalName" {
```

Comment:
I think it would be clearer if we instead checked if the type is ExternalName up front, either at the top of this function (with an immediate `return nil`) or in the calling function (don't call this function if ExternalName).

Reply:
I'm the one who opted for the alternative of a no-op but when I read it again I had the same thought. Apologies for misleading but this is a good minor change. I'll open a new tix.